Unable to read rows with null fields: "NullPointerException: Column ... contains NULL value"

I have a field mapped to nullable column in my entity class (using Kotlin).
```
        @Column(name = "field_a", spannerType = INT64, nullable = true)
        var fieldA: Short?,
```

_(I am aware that the `nullable = true` only applies to schema generation after reading https://github.com/spring-cloud/spring-cloud-gcp/issues/1078)_

I am able to save records with null values for this field. However, when I attempt to query the spanner table, I encounter an NPE:
```
Caused by: java.lang.NullPointerException: Column field_a contains NULL value
	at com.google.cloud.spanner.AbstractStructReader.checkNonNull(AbstractStructReader.java:336) ~[google-cloud-spanner-1.4.0.jar:1.4.0]
	at com.google.cloud.spanner.AbstractStructReader.checkNonNullOfType(AbstractStructReader.java:348) ~[google-cloud-spanner-1.4.0.jar:1.4.0]
	at com.google.cloud.spanner.AbstractStructReader.getLong(AbstractStructReader.java:113) ~[google-cloud-spanner-1.4.0.jar:1.4.0]
	at org.springframework.cloud.gcp.data.spanner.core.convert.StructAccessor.getSingleValue(StructAccessor.java:136) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.convert.StructPropertyValueProvider.readSingleWithConversion(StructPropertyValueProvider.java:104) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.convert.StructPropertyValueProvider.getPropertyValue(StructPropertyValueProvider.java:88) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.convert.StructPropertyValueProvider.getPropertyValue(StructPropertyValueProvider.java:38) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.data.mapping.model.PersistentEntityParameterValueProvider.getParameterValue(PersistentEntityParameterValueProvider.java:71) ~[spring-data-commons-2.1.2.RELEASE.jar:2.1.2.RELEASE]
	at org.springframework.data.convert.ClassGeneratingEntityInstantiator$EntityInstantiatorAdapter.extractInvocationArguments(ClassGeneratingEntityInstantiator.java:246) ~[spring-data-commons-2.1.2.RELEASE.jar:2.1.2.RELEASE]
	at org.springframework.data.convert.ClassGeneratingEntityInstantiator$EntityInstantiatorAdapter.createInstance(ClassGeneratingEntityInstantiator.java:219) ~[spring-data-commons-2.1.2.RELEASE.jar:2.1.2.RELEASE]
	at org.springframework.data.convert.ClassGeneratingEntityInstantiator.createInstance(ClassGeneratingEntityInstantiator.java:84) ~[spring-data-commons-2.1.2.RELEASE.jar:2.1.2.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.convert.ConverterAwareMappingSpannerEntityReader.read(ConverterAwareMappingSpannerEntityReader.java:92) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.convert.ConverterAwareMappingSpannerEntityProcessor.mapToList(ConverterAwareMappingSpannerEntityProcessor.java:79) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.SpannerTemplate.mapToListAndResolveChildren(SpannerTemplate.java:480) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
	at org.springframework.cloud.gcp.data.spanner.core.SpannerTemplate.query(SpannerTemplate.java:176) ~[spring-cloud-gcp-data-spanner-1.1.0.RELEASE.jar:1.1.0.RELEASE]
```

How do I go about reading null values in this case?

--- a/spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/StructPropertyValueProvider.java
+++ b/spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/StructPropertyValueProvider.java
@@ -82,6 +82,9 @@ class StructPropertyValueProvider implements PropertyValueProvider<SpannerPersis
 		if (!this.structAccessor.hasColumn(colName)) {
 			throw new SpannerDataException("Column not found: " + colName);
 		}
+		if (this.structAccessor.isNull(colName)) {
+			return null;
+		}
 		Class propType = spannerPersistentProperty.getType();
 		Object value = ConversionUtils.isIterableNonByteArrayType(propType)
 				? readIterableWithConversion(spannerPersistentProperty)

--- a/spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/domain/Trade.java
+++ b/spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/domain/Trade.java
@@ -65,8 +65,17 @@ public class Trade {
 	@Interleaved
 	private List<SubTrade> subTrades;
 
+	/**
+	 * Partial constructor. Intentionally tests a field that is left null sometimes.
+	 * @param symbol the symbol.
+	 */
+	public Trade(String symbol, List<Instant> executionTimes) {
+		this.symbol = symbol;
+		this.executionTimes = executionTimes;
+	}
+
 	public static Trade aTrade() {
-		Trade t = new Trade();
+		Trade t = new Trade("ABCD", new ArrayList<>());
 		String tradeId = UUID.randomUUID().toString();
 		String traderId = UUID.randomUUID().toString();
 
@@ -74,14 +83,12 @@ public class Trade {
 
 		t.tradeDetail.id = tradeId;
 		t.age = 8;
-		t.symbol = "ABCD";
 		t.action = "BUY";
 		t.traderId = traderId;
 		t.tradeTime = Instant.ofEpochSecond(333);
 		t.tradeDate = Date.from(t.tradeTime);
 		t.tradeDetail.price = 100.0;
 		t.tradeDetail.shares = 12345.6;
-		t.executionTimes = new ArrayList<>();
 		for (int i = 1; i <= 5; i++) {
 			t.executionTimes.add(Instant.ofEpochSecond(i));
 		}

--- a/spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/domain/Trade.java
+++ b/spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/domain/Trade.java
@@ -68,6 +68,7 @@ public class Trade {
 	/**
 	 * Partial constructor. Intentionally tests a field that is left null sometimes.
 	 * @param symbol the symbol.
+	 * @param executionTimes the list of execution times.
 	 */
 	public Trade(String symbol, List<Instant> executionTimes) {
 		this.symbol = symbol;

